### PR TITLE
More formatted console output

### LINKED ISSUE #764: Better visual design for error messages
Instead of printing error messages as plain text in the console, they could be formatted as HTML elements, kind of like a "cards" UI.  The goals would be:

1. Make them more visual and less technical looking.  For example, add a fun-looking icon for errors or warnings.
2. Make them more noticeable.
3. Separate them more from standard console output (like stdout, or debug tracing)
4. Maybe hide the extra info such as stack traces and "in the expression..." until the card is expanded?

### PR DESCRIPTION
**Pull request by nixorn**
Closes #764 

I took style for errors/log/warnings from firefox console.

### CODE REVIEW

File: web/js/run.js [hunk 1]
```diff
@@ -13,23 +13,14 @@
...
-                message.classList.add('error');
-            }
-
+        if (window.parent && window.parent.printMessage) {
+            window.parent.printMessage(type, str);
             return;
         }
     } catch (e) {}
-
-    console.log(str);
```
Comment (cdsmith) [inline web/js/run.js]: Please leave this here.  This is here to make sure output messages are still accessible somehow when the program run outside of the CodeWorld environment.

**Comment by cdsmith** [comment]
Thanks!  This looks great.  Left a few minor comments, and a couple bigger ones.  My main concerns are (a) getting this to work with Edge, which doesn't implement the details tag, and (b) not removing the overall styling for failed programs.

File: web/js/codeworld_shared.js [hunk 2]
```diff
@@ -1185,3 +1165,61 @@ function shareFolder_(mode) {
...
+    box.appendChild(messageGutter)
+    box.appendChild(messageContent);
+    outputDiv.appendChild(box);
+
+    if (splitted.length < 2) {
+        let singleLineMsg = document.createElement("div");
+        singleLineMsg.innerHTML = message
+        messageContent.appendChild(singleLineMsg)
+    } else {
+        let details = document.createElement("details"),
```
[the remark this answers is not in the document]
Comment (nixorn) [reply web/js/codeworld_shared.js]: Ok, I'll look what I can do. Need to setup Edge somewhere.

File: web/js/codeworld.js [hunk 3]
```diff
@@ -850,14 +850,17 @@ function run(hash, dhash, msg, error, generation) {
...
     let message = document.getElementById('message');
-    message.innerHTML = '';
-    addToMessage(msg);
-
-    if (error) {
-        message.classList.add('error');
-    } else {
-        message.classList.remove('error');
-    }
+    message.innerHTML = ''
```
[the remark this answers is not in the document]
Comment (nixorn) [reply web/js/codeworld.js]: Ok.

File: web/js/codeworld.js [hunk 4]
```diff
@@ -850,14 +850,17 @@ function run(hash, dhash, msg, error, generation) {
     }
 
     let message = document.getElementById('message');
-    message.innerHTML = '';
-    addToMessage(msg);
-
-    if (error) {
```
[the remark this answers is not in the document]
Comment (nixorn) [reply web/js/codeworld.js]: Ok.

File: web/js/run.js (hunk 1 above)
Comment (nixorn) [reply web/js/run.js]: Sorry, I assumed it was mine forgotten log :smile:

File: web/js/codeworld_shared.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (cdsmith) [reply web/js/codeworld_shared.js]: If you copy a polyfill script into the project, there is a process for that.  Place the script in `third_party`, in a new subdirectory with the script and the relevant LICENSE or COPYING file from the other project.  Then add a symlink from `web/js` to that file.

I am not so concerned with testing; if we trust the poyfill, don't waste too much time.  I have Edge on a computer here, so I can test after merging.

[the remark this answers is not in the document]
Comment (nixorn) [reply web/js/codeworld_shared.js]: Ok, trying elements from this list starting from https://github.com/javan/details-element-polyfill

**Pull request merged**